### DISCUSSION TITLE
Possibility of a v4.6.0 release to use the recent dynamic import refactors?

### DISCUSSION THREAD

**Discussion by itsame-luigi** [mikro-orm/mikro-orm#2338] [General]
I'm using mikro-orm in a project that uses both TypeScript 4.4 and ECMAScript modules. I'm managing migrations using `ts-node`, and I've found that the combination of `ts-node` + `{ "type": "module" }` (in package.json) doesn't work with mikro-orm v4.5.9.

To get things to work I'm using a custom wrapper around `@mikro-orm/cli` that patches `ConfigurationLoader.getconfiguration`, `Migrator.prototype.resolve`, and `MetadataDiscovery.prototype.initColumnType` (and others) to switch from using `require(path)` to `await import(path)`. I noticed that `master` already has these changes and I was wondering if there was a possibility of a v4.6.0 release with these updates.

**Comment by B4nan** [comment]
Not really, 1. master has those changes, but still compiles to CJS, so they will be compiled to requires anyway (iirc this should be possiblet to get around with TS 4.5), and 2. some of the changes were breaking (requires some APIs to become async). Also I believe there are still some dynamic requires in the codebase that were too hard to refactor.
